packages/terra-form-input/src/terra-dev-site/test/form-input/SupportedTypesInput.test.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import Input from '../../../Input';
+import styles from './common/Input.test.module.scss';
+
+const cx = classNames.bind(styles);
+
+const supportedTypes = [undefined, 'text', 'number', 'password', 'email', 'search', 'tel', 'url', 'hidden'];
```

Comment:
Should we exclude `undefined` as a supported type in this test since `undefined` is technically not an input type? This would actually show `type=undefined` in the DOM. If `type` was not set as a prop, then type would just be `undefined` and there would _not_ be a type attribute in the DOM since it would just be the default type of text as illustrated in the existing [default-form-input test example](https://engineering.cerner.com/terra-core/tests/terra-form-input/form-input/default-form-input).

![Screen Shot 2021-04-19 at 10 58 15 AM](https://user-images.githubusercontent.com/22618655/115267503-dd4ed300-a0fe-11eb-8be3-a4f207ea9ad2.png)

Reply:
it's not a type, but since Terra doesn't set the type to "text" as default (even though HTML does when it isn't provided), it was added here as an extra test case because the component can be generated without a type since it is undefined by default and still needs to work. I see now the bad effect - generating `type=undefined` is undesired.

Since other tests already check for when the type isn't set, let's remove it.

Reply:
Updated here https://github.com/cerner/terra-core/pull/3386/commits/6ccb549abcd88c1ad6d0af5ea0318a0d38da4652.